### PR TITLE
Enable deterministic builds

### PR DESCRIPTION
**Pull request by xPaw**
@JustArchi Archi do we need to add any other magic to improve the nuget?

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -17,11 +17,17 @@
...
         <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
         <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
         <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
     </PropertyGroup>
-</Project>
\ No newline at end of file
+
+    <PropertyGroup>
+        <Deterministic>true</Deterministic>
+        <ContinuousIntegrationBuild Condition="'$(CI)' == 'true'">true</ContinuousIntegrationBuild>
```
Comment (yaakov-h) [inline Directory.Build.props:31]: Isn’t this set automatically based on env cars these days?

Comment (xPaw) [reply Directory.Build.props:31]: I don't think it is, I checked other packages that had Deterministic set, but nuget analyzer still complained about this not being set.

**Comment by JustArchi** [comment]
@xPaw I only set deterministic to true in csproj, and add `-p:ContinuousIntegrationBuild=true` when building within CI, so pretty much what you did, but I prefer to not rely on `CI` being set in the env of the runners.

**Comment by xPaw** [comment]
CI env is documented here: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

It's also commonly available on other systems.

**Pull request merged**